### PR TITLE
Introduce Image Processing task type

### PR DESCRIPTION
**Pull request by Anghellia**

### CODE REVIEW

File: modelplace_api/utils.py [hunk 1]
```diff
@@ -17,6 +17,19 @@
...
 
+def is_numpy_array_equal(result: np.ndarray, gt: np.ndarray, error: float) -> bool:
+    '''
+    Applies element-wise comparison of two ndarrays and defines
+    whether the ratio of matched elements is eligible
+    '''
+    if result.size != gt.size:
+        raise RuntimeError('"result" and "gt" arrays must have equal size')
+    elements_num = result.size
+    equal_elements_num = np.equal(result, gt).sum()
```
Comment (PVSemk) [inline modelplace_api/utils.py:28]: I believe it's not suitable for float arrays' comparison. Probably, it's better to use something like np.isclose?

Comment (Anghellia) [reply modelplace_api/utils.py]: Updated, thanks!

**Review by PVSemk — APPROVED** [review]
LGTM

File: modelplace_api/utils.py [hunk 2]
```diff
@@ -17,6 +17,24 @@
...
+def is_numpy_array_equal(result: np.ndarray, gt: np.ndarray, error: float) -> bool:
+    '''
+    Applies element-wise comparison of two ndarrays and defines
+    whether the ratio of matched elements is eligible
+    '''
+    if result.size != gt.size:
+        raise RuntimeError('"result" and "gt" arrays must have equal size')
+    elements_num = result.size
+
+    if result.dtype == np.uint8:
```
Comment (lxdv) [inline modelplace_api/utils.py:29]: Why is `np.equal` used only for np.uint8? What about `np.uint16` or `32`?

File: modelplace_api/utils.py [hunk 3]
```diff
@@ -17,6 +17,24 @@
...
+        equal_elements_num = np.equal(result, gt).sum()
+    else:
+        equal_elements_num = np.isclose(result, gt, rtol=error).sum()
+
+    matching_ratio = equal_elements_num / elements_num
+    return matching_ratio >= 1 - error
```
Comment (lxdv) [inline modelplace_api/utils.py:35]: Minor: shorter

```python
def is_numpy_array_equal(result: np.ndarray, gt: np.ndarray, error: float) -> bool:
    '''
    Applies element-wise comparison of two ndarrays and defines
    whether the ratio of matched elements is eligible
    '''
    if result.size != gt.size:
        raise RuntimeError('"result" and "gt" shapes must be equal')

    if np.issubdtype(result.dtype, np.integer)
        matched = np.equal(result, gt).sum()
    else:
        matched = np.isclose(result, gt, rtol=error).sum()

    return matched / result.size >= 1 - error

File: modelplace_api/utils.py [hunk 4]
```diff
@@ -17,6 +17,24 @@
     from ._rle_mask import encode_binary_mask, decode_coco_rle
 
 
+def is_numpy_array_equal(result: np.ndarray, gt: np.ndarray, error: float) -> bool:
+    '''
+    Applies element-wise comparison of two ndarrays and defines
+    whether the ratio of matched elements is eligible
+    '''
+    if result.size != gt.size:
+        raise RuntimeError('"result" and "gt" arrays must have equal size')
```
Comment (lxdv) [inline modelplace_api/utils.py:26]: Seems you also need to make sure ndtypes of the arrays are the same

Comment (Anghellia) [reply modelplace_api/utils.py:26]: Added the check

Comment (lxdv) [reply modelplace_api/utils.py:26]: Also, I think better to compare `shape`s not `size`s as using `size`s won't raise an Exception however you'll get `ValueError: operands could not be broadcast together with shapes` in `np.equal` or `np.isclose`. You can check:

```python
import numpy as np

array = [1,2,3,4,5,6]

a = np.array(array).reshape((2,3))
b = np.array(array).reshape((3,2))

print(f"a.size==b.size is {a.size == b.size}")
print(f"a.shape==b.shape is {a.shape == b.shape}")

np.equal(a,b)
```

File: modelplace_api/utils.py [hunk 5]
```diff
@@ -17,6 +17,23 @@
...
+    if result.dtype != gt.dtype:
+        raise RuntimeError('"result" and "gt" arrays must have equal dtype')
```
Comment (lxdv) [inline modelplace_api/utils.py:28]: What do you think of using a bit changed message?
```python
raise RuntimeError(f'"result" and "gt" dtypes are different ({result.dtype},{gt.dtype}) - must be the same.')
```

**Review by lxdv — APPROVED** [review]
LGTM

**Pull request merged**